exe-common/src/network/peer.rs

```diff
@@ -29,6 +31,16 @@ impl Peer {
             config::net::Peer::Http(addr) => {
                 Ok(Peer::Http(hermes::HermesEndPoint::new(addr, network)))
             }
+            config::net::Peer::Ntt(addr) => {
+                let mut addrs_iter = addr.to_socket_addrs().unwrap();
+                match addrs_iter.next() {
+                    Some(addr) => ntt::NetworkCore::new(addr).map(Peer::Ntt),
+                    None => Err(Error::from(std::io::Error::new(
+                        std::io::ErrorKind::Other,
```

Comment:
I would say `ErrorKind::InvalidInput` fits better here. But even better is to have a case for this in `network::Error` rather than go through `std::io::Error`.

Reply:
I'm not sure I want to propagate all the errors in the rust-cardano, I'm not sure I'm ready for that endeavour.

Reply:
In fact, what's needed here is handling the error return of `to_socket_addrs()` above instead of dying on an invalid address.

Consider adding a variant `InvalidPeerAddress(String)` to `network::Error`; the address string that failed parsing goes inside the value.

Reply:
Fixed